### PR TITLE
adding custom filters MVP

### PR DESCRIPTION
**Pull request by PavelLinearB**
[workerB](https://workerb.linearb.io/v2/badge/collaboration-page?magicLinkId=ipiVJ1b)

### CODE REVIEW

File: docs/custom-filters-v1.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+!!! attention
+
+    :octicons-beaker-24: Coming soon
+
+
+###  Creating filters
+
+Filters can have at most one input parameter such as the available [context variables](context-variables.md), that is provided to the filter in the `.cm` file using a pipe.
+
```
Comment (vim-zz) [inline docs/custom-filters-v1.md:13]: Filters can have as many inputs arguemnts you wish. The first input argument can be "piped", when you have more than one you can use the more conventinoal function arguments, e.g. `f(input1, input2)` shall be used like `{{ input1 | f(input2) }}`

File: docs/custom-filters-v1.md [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+Filters must return a valid Javascript type (i.e Bool, Int, String, Object, etc...)
+
+In addition to the code implemetion of the filter, the filter creator must specify the npm dependecies in JSON format, as it appears in the `package_lock.json` file
```
Comment (vim-zz) [inline docs/custom-filters-v1.md:16]: This is something TBD, as we might want to limit imports to built-in imports etc.

File: docs/custom-filters-v1.md [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+automations:
+# approve the PR if the value of `myFilterOutput` is `true`
+  if:
+    - {{ myFilterOutput }}
+  run:
+    - action: approve@v1
+
+#assign the output of `myFilter` into a variable
+myFilterOutput: {{ pr | myFilter }}
+```
```
Comment (vim-zz) [inline docs/custom-filters-v1.md:47]: Consider simplifing the example by using the fitler in the first expression: 

```
automations:
# approve the PR if the value of `myFilter` is `true`
  if:
    - {{ pr | myFilter }}
  run:
    - action: approve@v1
```

**Review by vim-zz — COMMENTED** [review]
I added inline comments

File: docs/custom-filters-v1.md (hunk 2 above)
Comment (PavelLinearB) [reply docs/custom-filters-v1.md:16]: can't we add the other dependencies to our package_lock.json?

File: docs/custom-filters-v1.md (hunk 3 above)
Comment (PavelLinearB) [reply docs/custom-filters-v1.md:47]: I wanted to explicitly show that it can be assigned to a variable

File: docs/custom-filters-v1.md (hunk 2 above)
Comment (vim-zz) [reply docs/custom-filters-v1.md:16]: there might be few implications, this is why I suggest to have TBD

File: docs/custom-filters-v1.md (hunk 3 above)
Comment (vim-zz) [reply docs/custom-filters-v1.md:47]: assigning to variable is out of scope capability, avoiding using it will help keep the instructions more clear

**Pull request merged**